compute_sdk/globus_compute_sdk/sdk/executor.py

```diff
@@ -623,10 +666,12 @@ def _task_submitter_impl(self) -> None:
         # submission.  For example, if the user submitted to two different
         # endpoints, we separate the tasks by the dictionary key.
         SubmitGroupFutures = t.Dict[
-            t.Tuple[t.Optional[uuid.UUID], uuid.UUID], t.List[ComputeFuture]
+            t.Tuple[t.Optional[uuid.UUID], uuid.UUID, str],
+            t.List[ComputeFuture],
         ]
         SubmitGroupTasks = t.Dict[
-            t.Tuple[t.Optional[uuid.UUID], uuid.UUID], t.List[_TaskSubmissionInfo]
+            t.Tuple[t.Optional[uuid.UUID], uuid.UUID, str],
+            t.List[_TaskSubmissionInfo],
```

Comment:
Re-litigating [my comment on the previous PR in this area](https://github.com/funcx-faas/funcX/pull/1248#discussion_r1279564724), but I do still feel like the types here are unwieldy. I can imagine it only getting worse if/when we add more members to `submit_group`. My preference would be to see the key type separated out into a `namedtuple` or something similar, in order to make it more obvious what each member means.

That said, my sense of type-unwieldiness is probably more sensitive than others. I wouldn't be offended if y'all disagreed, or decided to punt on this until it becomes more of a headache.